pkg/domain/blueprintSpec.go

```diff
@@ -275,28 +298,45 @@ func (spec *BlueprintSpec) CheckEcosystemHealthAfterwards(healthResult ecosystem
 	}
 }
 
-func (spec *BlueprintSpec) StartApplying() (shouldApply bool) {
-	if spec.Config.DryRun {
-		spec.Events = append(spec.Events, BlueprintDryRunEvent{})
-	} else {
-		spec.Status = StatusPhaseInProgress
-		spec.Events = append(spec.Events, InProgressEvent{})
-		shouldApply = true
-	}
-	return
+// StartApplying marks the blueprint as in progress, which indicates, that the system started applying the blueprint.
+// This state is used to detect complete failures as this state will only stay persisted if the process failed before setting the state to blueprint applied.
+func (spec *BlueprintSpec) StartApplying() {
+	spec.Status = StatusPhaseInProgress
+	spec.Events = append(spec.Events, InProgressEvent{})
 }
 
-func (spec *BlueprintSpec) MarkFailed(err error) {
-	spec.Status = StatusPhaseFailed
+// MarkBlueprintApplicationFailed sets the blueprint state to application failed, which indicates that the blueprint could not be applied completely.
+// In reaction to this, further post-processing will happen.
+func (spec *BlueprintSpec) MarkBlueprintApplicationFailed(err error) {
+	spec.Status = StatusPhaseBlueprintApplicationFailed
 	spec.Events = append(spec.Events, ExecutionFailedEvent{err: err})
 }
 
+// MarkBlueprintApplied sets the blueprint state to blueprint applied, which indicates that the blueprint was applied successful and further steps can happen then.
 func (spec *BlueprintSpec) MarkBlueprintApplied() {
 	spec.Status = StatusPhaseBlueprintApplied
 	spec.Events = append(spec.Events, BlueprintAppliedEvent{})
 }
 
-func (spec *BlueprintSpec) MarkCompleted() {
-	spec.Status = StatusPhaseCompleted
-	spec.Events = append(spec.Events, CompletedEvent{})
+// CompletePostProcessing is used to mark the blueprint as completed or failed while the post-processing after trying to apply the blueprint.
+func (spec *BlueprintSpec) CompletePostProcessing() {
+	switch spec.Status {
+	case StatusPhaseEcosystemHealthyAfterwards:
+		spec.Status = StatusPhaseCompleted
+		spec.Events = append(spec.Events, CompletedEvent{})
+	case StatusPhaseInProgress:
+		spec.Status = StatusPhaseFailed
+		err := errors.New(handleInProgressMsg)
+		spec.Events = append(spec.Events, ExecutionFailedEvent{err: err})
+	case StatusPhaseEcosystemUnhealthyAfterwards:
+		spec.Status = StatusPhaseFailed
+		spec.Events = append(spec.Events, ExecutionFailedEvent{err: errors.New("ecosystem is unhealthy")})
+	case StatusPhaseBlueprintApplicationFailed:
+		spec.Status = StatusPhaseFailed
+		spec.Events = append(spec.Events, ExecutionFailedEvent{err: errors.New("could not apply blueprint")})
+	}
 }
+
+const handleInProgressMsg = "cannot handle blueprint in state " + string(StatusPhaseInProgress) +
```

Comment:
string-casting is unnecessary, since StatusPhase is already type string

Reply:
If you remove all `string(...)` the type of  `handleInProgressMsg` is also `StatusPhase` and cannot be used as a string elsewhere.
It seems that in go you cannot combine strings with other types represented by strings this way. :(